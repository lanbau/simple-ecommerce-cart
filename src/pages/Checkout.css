@media only screen and (max-width: 600px) {
    .single-checkout {
        flex-direction: column;
    }
}
.single-checkout {
    display: flex;
    justify-content: 'center';
}

.checkout-container {
    margin-top: 50px;
    margin-left: 20px;
    flex: 1;
    display: flex;
    /* align-items: center; */
    flex-direction: column;
}

.checkout-title {
    font-size: 32px;
    padding-bottom: 16px;
}