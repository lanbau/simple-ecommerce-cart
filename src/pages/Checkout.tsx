import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import './Checkout.css';
import { updateOrders, clearCart} from '../store/actions'
import { useHistory } from "react-router-dom";

interface BookInterface {
    body: string;
    id: number;
    price: number;
    src: string;
    title: string;
    userId: number;
}

interface RootState {
    cart: BookInterface[]
}
const Checkout = () => {
    const { cart } = useSelector((state: RootState) => state)
    const dispatch = useDispatch()
    const history = useHistory()
    const handleCheckout = () => {
        let newOrder = {
            date: new Date(),
            items: cart
        }
        dispatch(updateOrders(newOrder))
        dispatch(clearCart())
        history.push('/order')
    }

    const total = cart.reduce( (total, { price }) => {
        total = total + price
        return total
    }, 0)

    const [name, setName] = useState('')
    const [address, setAddress] = useState('')
    const [phone, setPhone] = useState('')

    const handleSubmit = (event: any) => {
        alert('A name was submitted: ' + name);
        event.preventDefault();
        const result = {
            name, 
            address,
            phone
        }
        console.log(result)
    }

    const handleChange = (event: any)  => {
        setName(event.target.value)
    }

    return <div className="single-checkout">
        <div className="checkout-container">
            <div className="checkout-title">Shipping Address</div>
            <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center' }}>
                <form onSubmit={handleSubmit} style={{ display: 'flex', flexDirection: 'column', marginBottom: 16 }}>
                    <label style={{ width: 150 }}>
                        Name:
                        <br></br>
                        <input type="text" onChange={e => setName(e.target.value)} />
                    </label>
                    <label>
                        Address:
                        <br></br>
                        <input type="text" onChange={e => setAddress(e.target.value)} />
                    </label>
                    <label>
                        Phone:
                        <br></br>
                        <input type="text" onChange={e => setPhone(e.target.value)} />
                    </label>
                    
                </form>
                <button className="book-save-button" onClick={handleSubmit} style={{ width: 150 }}>Save</button>
            </div>
            
        </div>
        <div className="checkout-container">
            <div className="checkout-title">Shopping Bag ({cart.length})</div>
            {cart.map(({ title, src, price }) => {
                return <div style={{ display: 'flex', alignItems: 'center', marginBottom: 8 }}>
                    <img src={src} style={{ width: 50, height: 50, marginRight: 8 }} />
                    <div style={{ display: 'flex', flexDirection: 'column', marginRight: 8 }}>
                        <div style={{ fontWeight: 'bold' }}>{`${title.substring(0, 30)}...`}</div>
                        <div>${price}</div>
                    </div>
                    {/* If have time */}
                    {/* <div style={{ fontWeight: 'bold', cursor: 'pointer' }}>
                        X
                    </div> */}
                </div>
            })
            }
            
        </div>
        <div className="checkout-container">
         <div className="checkout-title">Payment Info</div>
            <div style={{ marginBottom: 16 }}>Free Shipping & Tax Free</div>
            <div style={{ fontSize: 25, marginBottom: 16 }}>Total: ${total}</div>
            {/* Checkout will add cart to orders array with current Date */}
            <div style={{ display: 'flex' }}>
                {total > 0 && <div onClick={() => handleCheckout()} className="book-buy-button">Checkout</div> }
                {/* Return to Home */}
                {total > 0 && <div className="book-cancel-button" onClick={() => history.push('/')}>Cancel</div>}
            </div>
            
        </div>
    </div>
}
export default Checkout