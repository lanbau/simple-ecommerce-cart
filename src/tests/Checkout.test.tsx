import React from 'react'
import { useSelector } from 'react-redux'
import Checkout from '../pages/Checkout'
import renderer from 'react-test-renderer'

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}))

describe('Checkout Page', () => {
  test('renders cart', () => {
    useSelector.mockImplementation((selector) => selector({
      cart: [
        {
          userId: 1,
          id: 2,
          title: 'qui est esse',
          body: 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
          price: 49,
          src: 'https://picsum.photos/200/300?random=1'
        },
        {
          userId: 2,
          id: 2,
          title: 'qui est esse',
          body: 'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
          price: 49,
          src: 'https://picsum.photos/200/300?random=1'
        }
      ]
    }))
    const container = renderer
      .create(
        <Checkout />
      )
      .toJSON()
    expect(container).toMatchSnapshot()
  })
})