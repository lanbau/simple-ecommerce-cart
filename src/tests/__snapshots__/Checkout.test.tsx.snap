// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Page renders cart 1`] = `
<div
  className="single-checkout"
>
  <div
    className="checkout-container"
  >
    <div
      className="checkout-title"
    >
      Shipping Address
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <form
        onSubmit={[Function]}
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "marginBottom": 16,
          }
        }
      >
        <label
          style={
            Object {
              "width": 150,
            }
          }
        >
          Name:
          <br />
          <input
            onChange={[Function]}
            type="text"
          />
        </label>
        <label>
          Address:
          <br />
          <input
            onChange={[Function]}
            type="text"
          />
        </label>
        <label>
          Phone:
          <br />
          <input
            onChange={[Function]}
            type="text"
          />
        </label>
      </form>
      <button
        className="book-save-button"
        onClick={[Function]}
        style={
          Object {
            "width": 150,
          }
        }
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="checkout-container"
  >
    <div
      className="checkout-title"
    >
      Shopping Bag (
      2
      )
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginBottom": 8,
        }
      }
    >
      <img
        src="https://picsum.photos/200/300?random=1"
        style={
          Object {
            "height": 50,
            "marginRight": 8,
            "width": 50,
          }
        }
      />
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "marginRight": 8,
          }
        }
      >
        <div
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          qui est esse...
        </div>
        <div>
          $
          49
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginBottom": 8,
        }
      }
    >
      <img
        src="https://picsum.photos/200/300?random=1"
        style={
          Object {
            "height": 50,
            "marginRight": 8,
            "width": 50,
          }
        }
      />
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "marginRight": 8,
          }
        }
      >
        <div
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          qui est esse...
        </div>
        <div>
          $
          49
        </div>
      </div>
    </div>
  </div>
  <div
    className="checkout-container"
  >
    <div
      className="checkout-title"
    >
      Payment Info
    </div>
    <div
      style={
        Object {
          "marginBottom": 16,
        }
      }
    >
      Free Shipping & Tax Free
    </div>
    <div
      style={
        Object {
          "fontSize": 25,
          "marginBottom": 16,
        }
      }
    >
      Total: $
      98
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="book-buy-button"
        onClick={[Function]}
      >
        Checkout
      </div>
      <div
        className="book-cancel-button"
        onClick={[Function]}
      >
        Cancel
      </div>
    </div>
  </div>
</div>
`;
